src/Orleans/Lifecycle/ILifecycleObserver.cs

```diff
@@ -0,0 +1,22 @@
+﻿using System.Threading.Tasks;
+
+namespace Orleans
+{
+    /// <summary>
+    /// Lifecycle observer used to handle start and stop notification.
+    /// </summary>
+    public interface ILifecycleObserver
+    {
+        /// <summary>
+        /// Handle start notifications
+        /// </summary>
+        /// <returns></returns>
+        Task OnStart();
+
+        /// <summary>
+        /// Handle stop notifications
+        /// </summary>
+        /// <returns></returns>
+        Task OnStop();
+    }
```

Comment:
Although whether it is worthwhile to keep those info around is debatable. @jason-bragg probably has more say on this. For now I can see it may have value on debugging.